examples/petsc/src/swarmutils.c

```diff
@@ -282,14 +283,16 @@ PetscErrorCode DMSwarmInitalizePointLocations(DM dm_swarm, PointSwarmType point_
     } break;
     case SWARM_CELL_RANDOM: {
       // -- Set points randomly in each cell
-      PetscInt     dim = 3, num_cells_total = 1, num_cells[] = {1, 1, 1};
-      PetscScalar *point_coords;
+      PetscInt     dim = 3, num_cells_total = 1, num_cells[] = {1, 1, 1}, num_lower, num_upper;
```

Comment:
```suggestion
      PetscInt     dim = 3, num_cells_total = 1, num_cells[] = {1, 1, 1}, num_lower = 3, num_upper = 3;
```
to prevent non-deterministic
```
[0]PETSC ERROR: Argument out of range
[0]PETSC ERROR: n (-1439124880) cannot be negative
[0]PETSC ERROR: WARNING! There are unused option(s) set! Could be the program crashed before usage or a spelling mistake, etc!
[0]PETSC ERROR:   Option left: name:-ksp_monitor (no value) source: command line
[0]PETSC ERROR:   Option left: name:-ksp_view_solution value: cgns:s.cgns source: command line
[0]PETSC ERROR: See https://petsc.org/release/faq/ for trouble shooting.
[0]PETSC ERROR: Petsc Development GIT revision: v3.20.1-154-g1024d75c3cc  GIT Date: 2023-11-06 20:45:07 +0000
[0]PETSC ERROR: build/petsc-bpsswarm on a ompi named kichatna by jed Mon Dec  4 11:21:14 2023
[0]PETSC ERROR: Configure options --download-cgns --download-chaco --download-ctetgen --download-hpddm --download-ml --download-mumps --download-ptscotch=/home/jed/dl/scotch_7.0.4beta3.tar.gz --download-scalapack --download-superlu --download-triangle --with-cuda --with-debugging=0 --with-fortran-bindings=0 --with-hdf5 --with-hip=0 --with-metis --with-mpi-dir=/home/jed/usr/ccache/ompi --with-opengl --with-openmp --with-parmetis --with-strict-petscerrorcode --with-suitesparse --with-zlib COPTFLAGS="-O2 -march=native -ffp-contract=fast -g" PETSC_ARCH=ompi
[0]PETSC ERROR: #1 PetscOptionsScalarArray_Private() at /home/jed/petsc/src/sys/objects/aoptions.c:1669
[0]PETSC ERROR: #2 DMSwarmInitalizePointLocations() at /home/jed/src/libCEED/examples/petsc/src/swarmutils.c:294
[0]PETSC ERROR: #3 main() at /home/jed/src/libCEED/examples/petsc/bpsswarm.c:180
```

Reply:
Apparently there's a built-in cell-wise random coordinate function, which avoids some nasty issues with writing to the field in parallel.